EIPS/eip-6808.md

```diff
@@ -0,0 +1,526 @@
+
+```solidity
+function decreaseAllowance(address _spender, uint256 _subtractedValue) external returns (bool)
+```
+
+#### `transferFrom` function
+
+The function transfers `_amount` of tokens from address `_from` to address `_to`.
+
+The function MUST fire the `Transfer` event.
+
+The `transferFrom` method is used for a withdraw workflow, allowing contracts to transfer tokens on your behalf.
+The function SHOULD `revert` unless the `_from` account has deliberately authorized the sender.
+Each time the spender calls the function the contract subtracts and checks if the number of allowed transfers has reached 0,
+and when that happens the approval is revoked using an approve of 0 amount.
+
+**Note:** Transfers of 0 values MUST be treated as normal transfers and fire the `Transfer` event.
+
+```solidity
+function transferFrom(address _from, address _to, uint256 _amount) external returns (bool)
+```
+
+## Rationale
+
+The intent from individual technical decisions made during the development of **FKBTs** focused on maintaining consistency and backward compatibility with _ERC-20s_, all the while offering self-custodial security features to the user. It was important that **FKBTs** inherited all of the _ERC-20s_ characteristics to comply with requirements found in dApps which use fungible tokens on their platform. In doing so, it allowed for flawless backward compatibility to take place and gave the user the choice to decide if they want their **FKBTs** to act with **Default Behaviours**[^17]. We wanted to ensure that wide-scale implementation and adoption of **FKBTs** could take place immediately, without the greater collective needing to adapt and make changes to the already flourishing decentralized ecosystem.
+
+For developers and users alike, the `allowTransfer`[^3] and `allowApproval`[^4] functions both return bools on success and revert on failures. This decision was done purposefully, to keep consistency with the already familiar _ERC-20_. Additional technical decisions related to self-custodial security features are broken down and located within the security considerations section.
+
+## Backwards Compatibility
+
+**KBTs** are designed to be backward-compatible with existing token standards and wallets. Existing tokens and wallets will continue to function as normal, and will not be affected by the implementation of **FKBTs**.
+
+## Test Cases
+
+The [KBTstandard repository](../assets/eip-6808/README.md) has all the [tests](../assets/eip-6808/test/kbt20.js).
+
+Average Gas used (_GWEI_):
+
+- `addBindings` - 154,991
+- `resetBindings` - 30,534
+- `safeFallback` - 51,013
+- `allowTransfer` - 49,887
+- `allowApproval` - 44,971
+
+## Reference Implementation
+
+The GitHub repository [KBTstandard repository](../assets/eip-6808/README.md) contains the implementation. There's also a [diagram](../assets/eip-6808/Contract%20Interactions%20diagram.svg) with the contract interactions.
```

Comment:
Since this lives in the `assets` directory now, it's odd to mention the "KBTstandard repository".